src/intersection/compute_intersection.cc

```diff
@@ -296,26 +298,45 @@ bool ComputeIntersection<Simplex<1>, Simplex<2>>::compute(std::vector<Intersecti
     if(std::abs(w_sum) > rounding_epsilonX)
         w = w / w_sum;
     
-    // test whether all plucker products have the same sign
-    if(((w[0] > rounding_epsilon) && (w[1] > rounding_epsilon) && (w[2] > rounding_epsilon)) ||
-       ((w[0] < -rounding_epsilon) && (w[1] < -rounding_epsilon) && (w[2] < -rounding_epsilon))){
-        
+    unsigned int n_positive = 0;
+    unsigned int n_negative = 0;
+    unsigned int zero_idx_sum =0;
+
+    for (unsigned int i=0; i < 3; i++) {
+        if (w[i] > rounding_epsilon) n_positive++;
+        else if (w[i] < -rounding_epsilon) n_negative++;
+        else zero_idx_sum+=i;
+        //DebugOut().fmt("{} np: {} nn: {}\n", w[i], n_positive, n_negative);
+    }
+
+    if (n_positive>0 && n_negative>0) return false;
+
+    // test whether all plucker products are not zero
+    unsigned int n_both = n_positive + n_negative;
+    if (n_both > 0) {
         IntersectionPointAux<1,2> IP;
         
         if(compute_plucker(IP, w))
         {
+            // edge of triangle
+            if (n_both == 2) // one zero product
+                IP.set_topology_B(zero_idx_sum, 1);
+            else if (n_both == 1) // two zero products
+                IP.set_topology_B(RefElement<2>::oposite_node(3-zero_idx_sum), 0);
+
+            DebugOut().VarFmt(n_positive).VarFmt(n_negative);
```

Comment:
I do not get, how zero_idx_sum is index of triangle side.
The same with the oposite node and set_orientation is not obvious for me..